pkgs/build-support/setup-hooks/strip.sh

```diff
@@ -62,7 +62,12 @@ stripDirs() {
     if [ -n "${paths}" ]; then
         echo "stripping (with command $cmd and flags $stripFlags) in $paths"
         # Do not strip lib/debug. This is a directory used by setup-hooks/separate-debug-info.sh.
-        find $paths -type f -a '!' -wholename "$prefix/lib/debug/*" -exec $cmd $stripFlags '{}' \; 2>/dev/null
+        find $paths -type f -a '!' -path "$prefix/lib/debug/*" -print0 |
+            xargs -r -0 -n1 -P "$NIX_BUILD_CORES" $cmd $stripFlags || exit_code=$?
```

Comment:
This could be simplified to just this, right?

```bash
            xargs -r -0 -n1 -P "$NIX_BUILD_CORES" $cmd $stripFlags || [[ $exit_code = 123 ]
```

Reply:
I wonder even if we care about the exit code at all: do we care if all the files failed to strip? The old behavior did not:

```
$ find . -exec false \;; echo $?
0
```

Reply:
What Linus has here (and my suggestion also) _does_ match the old behaviour.  A 123 exit means that strip failed, and we are indeed ignoring that.

But any other error code from xargs means that xargs itself failed.  We don't want to ignore that, as it likely means we're holding it wrong.  Similarly, we weren't ignoring failures from `find` itself before.

Reply:
can't push

```
diff --git a/pkgs/build-support/setup-hooks/strip.sh b/pkgs/build-support/setup-hooks/strip.sh
index 8ee212be686..675cf2310a4 100644
--- a/pkgs/build-support/setup-hooks/strip.sh
+++ b/pkgs/build-support/setup-hooks/strip.sh
@@ -63,11 +63,12 @@ stripDirs() {
         echo "stripping (with command $cmd and flags $stripFlags) in $paths"
         # Do not strip lib/debug. This is a directory used by setup-hooks/separate-debug-info.sh.
         find $paths -type f -a '!' -path "$prefix/lib/debug/*" -print0 |
-            xargs -r -0 -n1 -P "$NIX_BUILD_CORES" $cmd $stripFlags || exit_code=$?
-        # xargs exits with status code 123 if some but not all of the
-        # processes fail. We don't care if some of the files couldn't
-        # be stripped, so ignore specifically this code.
-        [[ "$exit_code" = 123 || -z "$exit_code" ]]
+            xargs -r -0 -n1 -P "$NIX_BUILD_CORES" $cmd $stripFlags || [[ $? -eq 123 ]]
+            #                                                               ^
+            # xargs exits with status code 123 if some but not all of the
+            # processes fail. We don't care if some of the files couldn't
+            # be stripped, so ignore specifically this code.
+
         # 'strip' does not normally preserve archive index in .a files.
         # This usually causes linking failures against static libs like:
         #   ld: ...-i686-w64-mingw32-stage-final-gcc-13.0.0-lib/i686-w64-mingw32/lib/libstdc++.dll.a:
```

Reply:
https://github.com/NixOS/nixpkgs/pull/240734

Reply:
I intentionally didn't apply this because I find it a little clearer and the comment feels less awkward this way, sorry I forgot to write that.